pkg/image/api/types.go

```diff
@@ -60,6 +77,41 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureValidationResult holds a result of signature validation.
+type SignatureValidationResult string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob. Provided by caller.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side validation.
+	ValidatedOn unversioned.Time
+	// Result of validation performed at ValidatedOn.
+	ValidationResult SignatureValidationResult
+
+	// At most one of the following pointers can point to details. If the signature type isn't recognized,
+	// they will all be nil.
+
+	// Details for signature of type "atomic".
+	AtomicSignatureDetails *AtomicSignatureDetails
```

Comment:
> These are generic details core to the platform (image name).

No, these are specific to the signature format and the signature semantics.

The signature format explicitly calls this a “Docker reference” (for which I was told “image name” is the equivalent OpenShift term). If a future signature format will instead sign claims that an image identity is an URL, the signature would contain `"identity":{"some-kind-of-url":"$value"}` and the parsed contents of the signature could contain this URL, but definitely not in the “Atomic signature image name” field.